core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -339,20 +339,23 @@ class LogSegment(val log: FileMessageSet,
   }
 
   /**
-   * The time this segment has waited to be rolled. If the first message in the segment does not have a timestamp,
-   * the time is based on the create time of the segment. Otherwise the time is based on the timestamp of that message.
+   * The time this segment has waited to be rolled.
+   * If the first message has a timestamp we use the message timestamp to determine when to roll a segment.
```

Comment:
Would the following be clearer?

If the first message has a timestamp we use the message timestamp to determine when to roll a segment. A segment is rolled if the difference between the new message's timestamp and the first message's timestamp exceeds the segment rolling time.